squidb/src/com/yahoo/squidb/sql/TableStatement.java

```diff
@@ -50,21 +50,49 @@
     }
 
     private CompiledArgumentResolver compiledArgumentResolver = null;
+    private CompileContext compileContext = null;
 
+    /**
+     * Deprecated, use {@link #compile(CompileContext)} instead
+     */
     @Override
+    @Deprecated
     public final synchronized CompiledStatement compile(VersionCode sqliteVersion) {
+        return compile(CompileContext.defaultContextForVersionCode(sqliteVersion));
+    }
+
+    @Override
+    public final synchronized CompiledStatement compile(CompileContext compileContext) {
+        CompileContext contextToUse = this.compileContext != null ? this.compileContext : compileContext;
         if (compiledArgumentResolver == null) {
-            SqlBuilder builder = buildSql(sqliteVersion, true, false);
+            SqlBuilder builder = buildSql(contextToUse, true, false);
             compiledArgumentResolver = new CompiledArgumentResolver(builder);
         }
         return compiledArgumentResolver.resolveToCompiledStatement();
     }
 
+    /**
+     * Deprecated, use {@link #sqlForValidation(CompileContext)} instead
+     */
+    @Deprecated
     public final String sqlForValidation(VersionCode sqliteVersion) {
-        SqlBuilder builder = buildSql(sqliteVersion, true, true);
+        return sqlForValidation(CompileContext.defaultContextForVersionCode(sqliteVersion));
+    }
+
+    public final String sqlForValidation(CompileContext compileContext) {
+        SqlBuilder builder = buildSql(compileContext, true, true);
         return new CompiledArgumentResolver(builder).resolveToCompiledStatement().sql;
     }
 
+    /**
+     * Sets the {@link CompileContext} to be used when compiling this statement. If set, this value supersedes that
+     * provided by the {@link #compile(CompileContext)} method.
+     */
+    public void setCompileContext(CompileContext compileContext) {
```

Comment:
The ability to change the compile context on a one-off basis for a single statement was my original intention when I wrote this, but maybe it's not necessary or a good idea? We could always take it out and add it back in if there's ever demand for it.